app/components/metadata-panel.tsx

```diff
@@ -418,7 +422,33 @@ const TabPanelData = ({
   const [inputValue, setInputValue] = React.useState("");
 
   const options = React.useMemo(() => {
-    return dimensions.map((d) => ({ label: d.label, value: d }));
+    return dimensions.flatMap(
+      (
+        d
+      ): {
+        label: string;
+        value: Component;
+        isJoinByDimension: boolean;
+      }[] => {
+        if (
+          "isJoinByDimension" in d &&
+          d.isJoinByDimension &&
+          "originalIris" in d
```

Comment:
Nit: I think we have this check in several places now, maybe would make sense to extract to something like `isJoinByDimension`?

Reply:
Thanks, good idea, I extracted the guard in [`3e01901` (#1399)](https://github.com/visualize-admin/visualization-tool/pull/1399/commits/3e01901b169690c5e25a83fb324cd4089908b15c)